src/webnn/native/dml/ContextDML.h

```diff
@@ -0,0 +1,57 @@
+// Copyright 2022 The WebNN-native Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef WEBNN_NATIVE_DML_CONTEXT_DML_H_
+#define WEBNN_NATIVE_DML_CONTEXT_DML_H_
+
+#include "webnn/native/Context.h"
+
+#include "common/Log.h"
+#include "dml_platform.h"
+#include "webnn/native/Graph.h"
+
+namespace webnn::native::dml {
+
+    struct CommandRecorderDML {
```

Comment:
Can we move this into `.cpp`? This class will become very large (for InitializeOperator, ExecuteOperator, etc).

Reply:
You may mean InitializeGraph, ExecuteGraph, right? I think these will be in GraphDML.cpp.

Reply:
GraphDML would not record the actual commands into the command-list. It just builds the operator (graph) for the CommandRecorder. The CommandRecorder itself records these commands via InitializeGraph, ExecuteGraph, etc. This maintains a clear separation of duties.

Reply:
Yes, the commandlist is only used after the graph building. I guess your suggestion for separation may be:
CommandRecorder -> IntializeGraph()/ExecuteGraph() ? That's why you suggested defining the CommandRecorder or ExecutionContext?

Reply:
> Can we move this into `.cpp`? This class will become large-enough (for InitializeOperator, ExecuteOperator, etc).

Will consider re-organizing codes when we add more functions.

Reply:
SGTM.